src/engraving/rendering/dev/accidentalslayout.cpp

```diff
@@ -0,0 +1,1123 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#include <cfloat>
+
+#include "accidentalslayout.h"
+#include "layoutcontext.h"
+#include "../paddingtable.h"
+#include "shape.h"
+#include "tlayout.h"
+
+#include "dom/accidental.h"
+#include "dom/chord.h"
+#include "dom/ledgerline.h"
+#include "dom/note.h"
+#include "dom/score.h"
+
+using namespace mu::engraving;
+using namespace mu::engraving::rendering::dev;
+
+AccidentalsLayout::AccidentalsLayoutContext::AccidentalsLayoutContext(std::vector<Accidental*> acc,
+                                                                      std::vector<Chord*> ch)
+    : allAccidentals(std::move(acc)), chords(std::move(ch))
+{
+    sortTopDown(allAccidentals);
+    stackedAccidentals.reserve(allAccidentals.size());
+    stackedAccidentalsShape.elements().reserve(4 * allAccidentals.size());
+
+    initConstants();
+}
+
+void AccidentalsLayout::AccidentalsLayoutContext::initConstants()
+{
+    m_spatium = allAccidentals.front()->spatium();
+
+    // Hardcoded engraving parameters that don't have a style (some of these may get a style in future)
+    m_verticalAccidentalToAccidentalClearance = 0.15 * m_spatium;
+    m_verticalSharpToSharpClearance = 0.05 * m_spatium;
+    m_verticalAccidentalToChordClearance = 0.10 * m_spatium;
+    m_additionalPaddingForVerticals = 0.1 * m_spatium;
+    m_xPosSplitThreshold = 0.5 * m_spatium;
+    m_xVerticalAlignmentThreshold = 0.75 * m_spatium;
+    m_sharpAndNaturalLedgerLinePadding = 0.1 * m_spatium;
+    m_reducedFlatToNotePadding = 0.15 * m_spatium;
+    m_flatKerningOfFourth = -0.15 * m_spatium;
+    m_naturalKerningOfFourth = -0.05 * m_spatium;
+
+    const MStyle& style = allAccidentals.front()->style();
+    m_accidentalAccidentalDistance = style.styleMM(Sid::accidentalDistance);
+    m_orderFollowNoteDisplacement = style.styleB(Sid::accidentalOrderFollowsNoteDisplacement);
+    m_alignOctavesAcrossSubChords = style.styleB(Sid::alignAccidentalOctavesAcrossSubChords);
+    m_keepSecondsTogether = style.styleB(Sid::keepAccidentalSecondsTogether);
+    m_alignOffsetOctaves = style.styleB(Sid::alignOffsetOctaveAccidentals);
+}
+
+void AccidentalsLayout::layoutAccidentals(const std::vector<Chord*>& chords, LayoutContext& ctx)
+{
+    std::vector<Accidental*> allAccidentals;
+    std::vector<Accidental*> redundantAccidentals;
+
+    collectAccidentals(chords, allAccidentals, redundantAccidentals);
+
+    if (allAccidentals.empty()) {
+        return;
+    }
+
+    for (Accidental* acc : redundantAccidentals) {
+        acc->setPos(0.0, 0.0);
+        acc->setbbox(RectF());
+    }
+
+    for (Accidental* acc : allAccidentals) {
+        acc->setPos(0.0, 0.0);
+        acc->computeMag();
+        TLayout::layoutAccidental(acc, acc->mutldata(), ctx.conf());
+    }
+
+    AccidentalsLayoutContext accidentalsLayoutContext(std::move(allAccidentals), std::move(chords));
```

Comment:
std::move(chords) - this has no effect as chords is const std::vector<Chord*>& :)